contracts/mainnet/connectors/liquity/main.sol

```diff
@@ -0,0 +1,347 @@
+        _eventParam = abi.encode(msg.sender, amount, getId);
+    }
+
+    /**
+     * @dev Increase or decrease Trove ETH collateral and LUSD debt in one transaction
+     * @notice Adjust Trove debt and/or collateral
+     * @param maxFeePercentage The maximum borrow fee that this transaction should permit 
+     * @param withdrawAmount Amount of ETH to withdraw
+     * @param depositAmount Amount of ETH to deposit
+     * @param borrowAmount Amount of LUSD to borrow
+     * @param repayAmount Amount of LUSD to repay
+     * @param upperHint Address of the Trove near the upper bound of where the user's Trove will now sit in the ordered Trove list
+     * @param lowerHint Address of the Trove near the lower bound of where the user's Trove will now sit in the ordered Trove list
+     * @param getDepositId Optional storage slot to retrieve the ETH to deposit
+     * @param setWithdrawId Optional storage slot to store the withdrawn ETH to
+     * @param getRepayId Optional storage slot to retrieve the LUSD to repay
+     * @param setBorrowId Optional storage slot to store the LUSD borrowed
+    */
+    function adjust(
+        uint maxFeePercentage,
+        uint withdrawAmount,
+        uint depositAmount,
+        uint borrowAmount,
+        uint repayAmount,
+        address upperHint,
+        address lowerHint,
+        uint getDepositId,
+        uint setWithdrawId,
+        uint getRepayId,
+        uint setBorrowId
+    ) external payable returns (string memory _eventName, bytes memory _eventParam) {
+        AdjustTrove memory adjustTrove;
+
+        adjustTrove.maxFeePercentage = maxFeePercentage;
+        adjustTrove.withdrawAmount = withdrawAmount;
+        adjustTrove.depositAmount = getUint(getDepositId, depositAmount);
+        adjustTrove.borrowAmount = borrowAmount;
+        adjustTrove.repayAmount = getUint(getRepayId, repayAmount);
+        adjustTrove.isBorrow = borrowAmount > 0;
+
+        borrowerOperations.adjustTrove{value: depositAmount}(
+            adjustTrove.maxFeePercentage,
+            adjustTrove.withdrawAmount,
+            adjustTrove.borrowAmount,
+            adjustTrove.isBorrow,
+            upperHint,
+            lowerHint
+        );
+        
+        // Allow other spells to use the withdrawn collateral
+        setUint(setWithdrawId, withdrawAmount);
+
+        // Allow other spells to use the borrowed amount
+        setUint(setBorrowId, borrowAmount);
+
+        _eventName = "LogAdjust(address,uint,uint,uint,uint,uint,uint,uint,uint,uint)";
+        _eventParam = abi.encode(msg.sender, maxFeePercentage, depositAmount, borrowAmount, getDepositId, setWithdrawId, getRepayId, setBorrowId);
+    }
+
+    /**
+     * @dev Withdraw remaining ETH balance from user's redeemed Trove to their DSA
+     * @notice Claim remaining collateral from Trove
+    */
+    function claimCollateralFromRedemption() external returns(string memory _eventName, bytes memory _eventParam) {
```

Comment:
Shouldn’t we have here a `setUint` for the collateral obtained?

Reply:
https://github.com/liquity/dsa-connectors/commit/97db94fe595330743385f26c30e15ab28cc18063#r645495186